### ISSUE TITLE
Fails to backup

### ISSUE THREAD

**Issue by sethgodin1** [rmcrackan/Libation#854]
**Describe the bug**
Running on a Macbook pro under 14.3.1

I'm able to see my library. I hit 'liberate visible books'

it seems to fail.

I might be doing it wrong!
![Screenshot 2024-02-21 at 7 56 00 AM](https://github.com/rmcrackan/Libation/assets/81632168/a946dc54-1f22-44c1-a672-8a22ec8ff372)

here's the log
[libation log.txt](https://github.com/rmcrackan/Libation/files/14359818/libation.log.txt)

**Comment by rmcrackan** [comment]
Sorry about this. Can you go to Settings > Settings... and click the "Open log folder" at the bottom. That should open the folder which has files like "Log202401.zip" which will hopefully have some more technical details for me. Could you attach that to this ticket?

**Comment by sethgodin1** [comment]
I'm super grateful for the work you do and the fast response too. Thank you!
[_Log202402.zip](https://github.com/rmcrackan/Libation/files/14360163/_Log202402.zip)

I'm guessing it's user error

**Comment by rmcrackan** [comment]
@sethgodin1 Thank you so much for this log file. This may be a bug with mp3 on mac specifically. Can you try 1 book again not as mp3? You can right click the book to try just that one again.

@Mbucari Would you mind taking a look at this?

```
System.PlatformNotSupportedException: Could not load ffmpegaac or ffmpegaac.arm64.dylib
   at AAXClean.Codecs.Mp4FileExtensions.DllImportResolver(String libraryName, Assembly assembly, Nullable`1 searchPath) in /home/runner/work/AAXClean.Codecs/AAXClean.Codecs/src/AAXClean.Codecs/Mp4FileExtensions.cs:line 36
```

**Comment by sethgodin1** [comment]
screenshots attached
![Screenshot 2024-02-21 at 8 55 47 AM](https://github.com/rmcrackan/Libation/assets/81632168/12c37a56-d218-45ed-9b7e-0fcc6538e4bf)
![Screenshot 2024-02-21 at 8 55 27 AM](https://github.com/rmcrackan/Libation/assets/81632168/45fd6339-2aa8-403c-bee9-de404b32b357)

**Comment by Mbucari** [comment]
@sethgodin1 Can you please go into the Libation app bundle (right-click it then click "Show Package Contents"). Do you see a file in there named `ffmpegaac.arm64.dylib`?

**Comment by sethgodin1** [comment]
inside the macos folder in the Contents folder, I see as attached

![Screenshot 2024-02-21 at 12 07 06 PM](https://github.com/rmcrackan/Libation/assets/81632168/59d090e3-4fd1-4ea1-ba06-9869227d820d)

**Comment by rmcrackan** [comment]
@Mbucari you mentioned arm64 and he has x64. I don't know mac -- is this just a matter of using the other download type?

**Comment by rmcrackan** [comment]
Related bug #1020

**Comment by Mbucari** [comment]
> @Mbucari you mentioned arm64 and he has x64. I don't know mac -- is this just a matter of using the other download type?

No, the build scripts will include the correct lib for each targeted platform.

**Comment by Mbucari** [comment]
> you mentioned arm64 and he has x64. I don't know mac -- is this just a matter of using the other download type?

@rmcrackan 
I think you might be onto something. AAXClean.Codecs loads the native libraries at runtime. [Here's](https://github.com/Mbucari/AAXClean.Codecs/blob/45c3cd40159c8e411acfdac4470f3220c2d52066/src/AAXClean.Codecs/Mp4FileExtensions.cs#L19-L37) where it tries to find the dll by name. Notice Line #23 where it determines the architecture to load by getting the _OS_ architecture. 

Here's what I think is happening. The users's machines are running on Apple Silicon which is arm64, but they downloaded the x64 packages. x64 will run on apple silicon under Apple's Rosetta translation layer, but AAXClean is trying to load load the library of the System's architecture, not the Libation package's architecture.

A temporary fix is to have the users delete their existing Libation app bundle and download the arm64 bundle.

I [edited the macos build script](https://github.com/rmcrackan/Libation/commit/0a749d2d88fb62a3223e307cd13365d1440bd2a5) to rename the native lib files, removing the architecture name. This should work because AAXClean tries to load both "ffmpegaac.[ARCH].dylib" AND just "ffmpegaac". Renaming the lib to just "ffmpegaac.dylib" should cause the system's library resolver to find it (and the same goes with [the libmp3lame resolver](https://github.com/Mbucari/AAXClean.Codecs/blob/45c3cd40159c8e411acfdac4470f3220c2d52066/src/NAudio.Lame/LameDLLWrap/NativeMethods.cs#L60-L78)

**Comment by rmcrackan** [comment]
@Mbucari Gracias. I'll create a pre-release soon

**Comment by rmcrackan** [comment]
Please try this pre-release and let me know if it helps your issue : https://github.com/rmcrackan/Libation/releases/tag/v11.5.2

**Comment by rmcrackan** [comment]
This bug seems to be out of date. If you have more information, please reply here and we can look into it further.

Please try the latest version: https://github.com/rmcrackan/Libation/releases/latest
